File: almost_fibonacci.cpp
#include <bits/stdc++.h>
#define endl "\n"
#define int long
#define long long long
const int MOD = (int)1e9 + 7;
const int limit = 1000001;
using namespace std;


long fibo(int n) {
    if (n == 0) return 0;
    if (n == 1) return 1;

    vector<long> dp(n + 1);
    dp[0] = 0;
    dp[1] = 1;

    for (long i = 2; i <= n; i++) {
        dp[i] = (dp[i - 1] + dp[i - 2] + 1) % MOD;
    }

    return dp[n];
}

main(){
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);
    long n;
    cin >> n;
    long res = fibo(n);
    cout << res << endl;

    return 0;
}

File: almost_fibonacci.py
MOD = 1000000007

def fibo(n):
    if n == 1:
        a = 1
        b = 1
    else:
        x, y = fibo(n // 2)
        a = ((x % MOD) * (x % MOD) + (y % MOD) * (y % MOD)) % MOD  # a = x*x + y*y
        b = (2 * (x % MOD) * (y % MOD) - (y % MOD) * (y % MOD)) % MOD  # b = 2*x*y - y*y
        if n % 2:
            a = a + b  # a = a + b
            b = a - b  # b = a - b
    return a, b

n = int(input())

if n == 0:
    a = 1
else:
    a, b = fibo(n + 1)

print((a - 1 + MOD) % MOD)